tools/Custom/JsonExtensions.cs

```diff
@@ -1,62 +1,85 @@
-        /// </returns>
+        /// <param name="token">The JToken (JObject or JArray) to process.</param>
+        /// <returns>The cleaned JSON string with "defaultnull" values removed and "null" strings converted to null.</returns>
         /// <example>
         /// JObject json = JObject.Parse(@"{""name"": ""John"", ""email"": ""defaultnull"", ""address"": ""null""}");
         /// string cleanedJson = json.RemoveDefaultNullProperties();
         /// Console.WriteLine(cleanedJson);
         /// // Output: { "name": "John", "address": null }
         /// </example>
-        public static string RemoveDefaultNullProperties(this JObject jsonObject)
+        public static string RemoveDefaultNullProperties(this JToken token)
         {
             try
             {
-                foreach (var property in jsonObject.Properties().ToList())
+                if (token is JObject jsonObject)
                 {
-                    if (property.Value.Type == JTokenType.Object)
+                    foreach (var property in jsonObject.Properties().ToList())
                     {
-                        RemoveDefaultNullProperties((JObject)property.Value);
-                    }
-                    else if (property.Value.Type == JTokenType.Array)
-                    {
-                        foreach (var item in property.Value)
+                        if (property.Value.Type == JTokenType.Object)
                         {
-                            if (item.Type == JTokenType.Object)
-                            {
-                                RemoveDefaultNullProperties((JObject)item);
-                            }
+                            RemoveDefaultNullProperties(property.Value);
+                        }
+                        else if (property.Value.Type == JTokenType.Array)
+                        {
+                            RemoveDefaultNullProperties(property.Value);
+                        }
+                        else if (property.Value.Type == JTokenType.String && property.Value.ToString() == "defaultnull")
+                        {
+                            property.Remove();
+                        }
+                        else if (property.Value.Type == JTokenType.String && property.Value.ToString() == "null")
+                        {
+                            property.Value = JValue.CreateNull();
                         }
                     }
-                    else if (property.Value.Type == JTokenType.String && property.Value.ToString() == "defaultnull")
-                    {
-                        property.Remove();
-                    }
-                    else if (property.Value.Type == JTokenType.String && (property.Value.ToString() == "null"))
+                }
+                else if (token is JArray jsonArray)
+                {
+                    // Process each item in the JArray
+                    for (int i = jsonArray.Count - 1; i >= 0; i--)
```

Comment:
Àny specific reason to iterate this from the end?

Reply:
It's just a personal practice to ensure that the property is deleted safely. i.e, to prevent any possibility of skipping.